File: ruby/gmsec4/dox/StringField.rb
# Copyright 2007-2021 United States Government as represented by the
# Administrator of The National Aeronautics and Space Administration.
# No copyright is claimed in the United States under Title 17, U.S. Code.
# All Rights Reserved.

# This file is used for generating Ruby documentation; it only contains
# stubs of constructors and methods associated with the class.
#


module Libgmsec_ruby #:nodoc: don't document this

	# This is a specialized Field class that can be used to store a string value.
	#
	# ==== Example Usage
	#
	#    require 'libgmsec_ruby'
	#
	#    begin
	#        field = Libgmsec_ruby::StringField.new("MY-STR-FIELD", "Hello Ruby")
	#    end
	#
	class StringField < Libgmsec_ruby::Field
		# Constructor.
		#
		# ==== Attributes
		#
	    # * +name+  - The name of the binary field.
	    # * +value+ - The string value of the field.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if the name is nil, or is an empty string.
		# A GmsecException is thrown if the value is nil.
		#
		def initialize(name, value)
		end


		# Returns the value associated with the field.
		#
		# ==== Returns
		#
		# A string value.
		#
		def get_value
		end
	end
end

File: nodejs/gmsec5/dox/StringField.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file StringField.java
 */

package gmsec;


/**
 * Specialized Field class that can be used to retain a string value.
 */
public class StringField extends Field
{
	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data String data to associate with the field.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws SWIG_Exception Thrown if the data string is null.
	 * @throws GmsecException Thrown if the field name string is an empty string or is otherwise non-compliant.
	 */
	public StringField(name, data);


	/**
	 * Constructor.
	 *
	 * @param name      Name of the field.
	 * @param data      String data to associate with the field.
	 * @param is_header Used to indicate if Field is a header field.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws SWIG_Exception Thrown if the data string is null.
	 * @throws GmsecException Thrown if the field name string is an empty string or is otherwise non-compliant.
	 */
	public StringField(name, data, is_header);


	/**
	 * Copy constructor.
	 *
	 * @param other The other StringField object to copy.
	 *
	 * @throws SWIG_Exception Thrown if the given StringField object is null.
	 */
	public StringField(other);


	/**
	 * Returns the data stored within the StringField.
	 *
	 * @return A string value.
	 */
	public getValue();
}
